### DISCUSSION TITLE
i have an error while compiling

### DISCUSSION THREAD

**Discussion by enghady-ahmed** [luc-github/ESP3D#692] [Questions]
i remove my Arduino ID 1.8.16 and remove all file related to arduino to install the requested Arduino: 1.8.9 and ESP8266 2.5.2 
and after that i got below error while compiling  

Arduino: 1.8.9 (Windows 10), Board: "Arduino/Genuino Uno"

asyncwebserver.cpp:21:22: error: pgmspace.h: No such file or directory

compilation terminated.

exit status 1
pgmspace.h: No such file or directory

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by enghady-ahmed** [comment]
my problem solved after i select board as i missing select it code work and i will connect it to my printer , but the problem i use homemade printer using home made board with ATmega 664 so i think i will connect it to it's TX and RX

**Reply by luc-github** [reply]
asyncwebserver  is no well supported by esp3d you should not enable it

**Reply by enghady-ahmed** [reply]
but how i enable and disable it

**Reply by luc-github** [reply]
I just refer to your error message, check your config.h to be sure it is not enabled

**Comment by enghady-ahmed** [comment]
Dear Sir

the code is working on my old 3D printer it use ATmega644 and i use NodeMCu esp8266 to connect with it i really happy and thankful for you just want to know why temperature wasn't updated in graph although i receive it in command text
![temp](https://user-images.githubusercontent.com/69543004/141649681-6cd0678b-8657-48a5-b66c-24b6768ec151.JPG)

**Comment by luc-github** [comment]
same question as posted in webui; https://github.com/luc-github/ESP3D-WEBUI/discussions/205
